temporalcli/commands.server.go

```diff
@@ -73,6 +73,9 @@ func (t *TemporalServerStartDevCommand) run(cctx *CommandContext, args []string)
 		}
 		if opts.UIPort == 0 {
 			opts.UIPort = t.Port + 1000
+			if opts.UIPort > 65535 {
+				opts.UIPort = 65535
+			}
```

Comment:
This increases the probability of 65535 being chosen randomly (vs. other port numbers). Can we instead do something like:

```suggestion
			if opts.UIPort > 65535 {
				opts.UIPort = (t.Port - 1024) % (65536 - 1024) + 1024
			}
```

(why keep the if-statement at all? To preserve backward compatibility with the existing `+1000` behavior)